openedx/core/djangoapps/credentials/tests/test_utils.py

```diff
@@ -0,0 +1,114 @@
+        self.user = UserFactory()
+        self.url = credential_config.internal_api_url + 'user_credentials/?username=' + self.user.username
+
+    @httpretty.activate
+    def test_get_user_credentials(self):
+        """Verify user credentials data can be retrieve."""
+        self.mock_api(self.url, self.CREDENTIALS_API_RESPONSE)
+
+        actual = get_user_credentials(self.user)
+        self.assertEqual(actual, self.CREDENTIALS_API_RESPONSE['results'])
+
+    def test_get_user_credentials_credentials_disabled(self):
+        """Verify behavior when Credentials is disabled."""
+        self.create_credential_config(enabled=False)
+
+        actual = get_user_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @mock.patch('edx_rest_api_client.client.EdxRestApiClient.__init__')
+    def test_get_credentials_client_initialization_failure(self, mock_init):
+        """Verify behavior when API client fails to initialize."""
+        mock_init.side_effect = Exception
+
+        actual = get_user_credentials(self.user)
+        self.assertEqual(actual, [])
+        self.assertTrue(mock_init.called)
+
+    @httpretty.activate
+    def test_get_user_credentials_retrieval_failure(self):
+        """Verify behavior when data can't be retrieved from Credentials."""
+        self.mock_api(self.url, self.CREDENTIALS_API_RESPONSE, status_code=500)
+
+        actual = get_user_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    def test_get_user_program_credentials_issuance_disable(self):
+        """Verify that user program credentials cannot be retrieved if issuance is disabled."""
+        self.create_credential_config(enable_learner_issuance=False)
+        actual = get_user_program_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @httpretty.activate
+    def test_get_user_program_credentials_no_credential(self):
+        """Verify behavior if no credential exist."""
+        self.mock_api(self.url, {'results': []})
+        actual = get_user_program_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @httpretty.activate
+    def test_get_user_program_credentials_revoked(self):
+        """Verify behavior if credential revoked."""
+        credential_data = {"results": [
+            {
+                "id": 1,
+                "username": "test",
+                "credential": {
+                    "credential_id": 1,
+                    "program_id": 1
+                },
+                "status": "revoked",
+                "uuid": "dummy-uuid-1"
+            }
+        ]}
+        self.mock_api(self.url, credential_data)
+        actual = get_user_program_credentials(self.user)
+        self.assertEqual(actual, [])
+
+    @httpretty.activate
+    def test_get_user_program(self):
+        """Verify program credentials data can be retrieved and parsed correctly."""
+        credentials_config = self.create_credential_config()
```

Comment:
This code is extremely dense and hard to read. Please add some newlines to make it more legible.